src/publisher/middleware.py

```diff
@@ -277,3 +278,95 @@ def middleware(request):
         return response
 
     return middleware
+
+
+#
+#
+#
+
+
+def v3_vor_valid_under_v2(ajson):
+    "returns True if given article-json is valid under both version 2 and version 3 of the VOR spec"
+    if ajson["status"] != "vor":
+        return False
+
+    path_list = [
+        ("authorResponse", "doi"),
+        ("decisionLetter", "doi"),
+        ("digest", "doi"),
+    ]
+    for bit1, bit2 in path_list:
+        if bit1 in ajson:
+            if bit2 not in ajson[bit1]:
+                return False
+    return True
+
+
+def incompatible_v2_check(get_response_fn):
+    def middleware(request):
+        request_accept_header = request.META.get("HTTP_ACCEPT", "*/*")
+        response = get_response_fn(request)
+
+        if response.status_code != 200:
+            # unsuccessful response, ignore
+            return response
+
+        # are we returning VOR content?
+        vor_ctype = "application/vnd.elife.article-vor+json"
+        resp_ctype = get_content_type(response)
+        if vor_ctype not in resp_ctype:
+            return response  # nope, ignore
+
+        client_accepts_list = flatten_accept(request_accept_header)
+        client_accepts_vor_list = [
+            row for row in client_accepts_list if row[0] == vor_ctype
+        ]
+
+        # are they flexible in what they accept?
+        anything = ("*/*", "version", None)
+        almost_anything = ("application/*", "version", None)
+        any_vor = ("application/vnd.elife.article-vor+json", "version", None)
```

Comment:
these conditions look a bit specific, I wonder if they should just be the default case, with `if v3_vor_valid_under_v2()` being inside the opposite conditions; seems easier to check whether v1/v2 is being requested than all the other possible, flexible cases?